src/ipc/ipc4/helper.c

```diff
@@ -681,19 +700,19 @@ int ipc_comp_disconnect(struct ipc *ipc, ipc_pipe_comp_connect *_connect)
 	 * IPC or IDC task getting preempted which could result in buffers being only half connected
 	 * when a pipeline task gets executed.
 	 */
-	ll_block(cross_core_unbind);
+	ll_block(cross_core_unbind, flags);
 
 	if (cross_core_unbind) {
 #if CONFIG_CROSS_CORE_STREAM
 		/* Make sure LL has finished on both cores */
 		if (!cpu_is_me(src->ipc_config.core))
 			if (ll_wait_finished_on_core(src) < 0) {
-				ll_unblock(cross_core_unbind);
+				ll_unblock(cross_core_unbind, flags);
 				return IPC4_FAILURE;
 			}
 		if (!cpu_is_me(sink->ipc_config.core))
 			if (ll_wait_finished_on_core(sink) < 0) {
-				ll_unblock(cross_core_unbind);
+				ll_unblock(cross_core_unbind, flags);
 				return IPC4_FAILURE;
 			}
 #else
```

Comment:
@marcinszkudlinski Can you check the CI build fail? I think you missed a few spots in helper.c wheere CONFIG_CROSS_CORE_STREAM is ifdeffed out and there's call to ll_block/ll_unblock.